pkg/network/sriov_test.go

```diff
@@ -0,0 +1,156 @@
+		_validateSriov := func(
+			sriovRequested bool,
+			openshiftNetworkOperatorRunning bool,
+			openshiftSriovEnabled bool,
+		) []error {
+			conf := &opv1alpha1.NetworkAddonsConfigSpec{}
+			if sriovRequested {
+				conf.Sriov = &opv1alpha1.Sriov{}
+			}
+
+			var openshiftNetworkConfig *osv1.Network
+			if openshiftNetworkOperatorRunning {
+				openshiftNetworkConfig = &osv1.Network{}
+				if openshiftSriovEnabled {
+					openshiftNetworkConfig.Spec.AdditionalNetworks = append(openshiftNetworkConfig.Spec.AdditionalNetworks, osv1.AdditionalNetworkDefinition{
+						Type:         osv1.NetworkTypeRaw,
+						Name:         "sriov-network-name",
+						RawCNIConfig: "{\"type\": \"sriov\"}",
+					})
+				}
+			}
+
+			return validateSriov(conf, openshiftNetworkConfig)
+		}
+
+		Context("when configuration is empty", func() {
+			sriovRequested := false
+			openshiftNetworkOperatorRunning := false
+			openshiftSriovEnabled := false
+			It("should pass", func() {
+				errorList := _validateSriov(sriovRequested, openshiftNetworkOperatorRunning, openshiftSriovEnabled)
+				Expect(errorList).To(BeEmpty())
+			})
+		})
+
+		Context("when openshfit network operator is running", func() {
+			openshiftNetworkOperatorRunning := true
+			Context("and has sriov enabled", func() {
+				openshiftSriovEnabled := true
+				Context("and configuration is empty", func() {
+					sriovRequested := false
+					It("should pass", func() {
+						errorList := _validateSriov(sriovRequested, openshiftNetworkOperatorRunning, openshiftSriovEnabled)
+						Expect(errorList).To(BeEmpty())
+					})
+				})
+				Context("and configuration requests sriov", func() {
+					sriovRequested := true
+					It("should fail with an error message", func() {
+						errorList := _validateSriov(sriovRequested, openshiftNetworkOperatorRunning, openshiftSriovEnabled)
+						Expect(len(errorList)).To(Equal(1), "validation failed due to an unexpected error: %v", errorList)
+						Expect(errorList[0].Error()).To(Equal("SR-IOV has been requested, but it's not compatible with OpenShift Cluster Network Operator SR-IOV support"))
+					})
+				})
+			})
+
+			Context("and has sriov disabled", func() {
+				openshiftSriovEnabled := false
+				Context("and configuration is empty", func() {
+					sriovRequested := false
+					It("should pass", func() {
+						errorList := _validateSriov(sriovRequested, openshiftNetworkOperatorRunning, openshiftSriovEnabled)
+						Expect(errorList).To(BeEmpty())
+					})
+				})
+				Context("and configuration requests sriov", func() {
+					sriovRequested := true
+					It("should pass", func() {
+						errorList := _validateSriov(sriovRequested, openshiftNetworkOperatorRunning, openshiftSriovEnabled)
+						Expect(errorList).To(BeEmpty())
+					})
+				})
+			})
+		})
+
+		Context("when openshfit network operator is not running", func() {
+			openshiftNetworkOperatorRunning := false
+			openshiftNetworkOperatorDisableMultiNetwork := false
```

Comment:
the name of this variable differs from what is used above, and it doesn't seem like matching the meaning of the helper function parameter (?): the function parameter means "whether openshift sriov network is enabled" while the variable name suggests the meaning is "whether multi-network support (?) is disabled"?

Reply:
I see, I instantiated it just so there is a value, it doesn't matter whether it's true or false, since operator is disabled. Adding a comment.

openshift's Network type has attribute DisableMultiNetwork. I don't like negative parameters, so I'm trying to hide it from upper level (eyes of a developer running these tests).

Reply:
it seems to me that the variable name doesn't belong here and was copy pasted from another test module (for multus).

Reply:
Should be better now